File: common/StudioFundsCalculator/StudioFundsCalculator.c
#include <stdio.h>

int calculateYearsUntilTarget(double initialCapital, double annualRate, double membershipFee, double targetAmount) {
    if (targetAmount <= initialCapital) {
        return 0; // If the target amount is less than or equal to the initial capital, return 0 years
    }

    double currentCapital = initialCapital;
    int year = 0;

    while (currentCapital < targetAmount) {
        year++;
        // Calculate end-of-year capital
        currentCapital = currentCapital * (1 + annualRate) + membershipFee * year;
        printf("%d年末 的金额: %.2f\n", year, currentCapital);
    }

    return year;
}

int main() {
    double initialCapital = 100; // Initial capital in ten thousand
    double annualRate = 0.10;    // Annual interest rate of 10%
    double membershipFee = 2;     // Membership renewal fee in ten thousand, increasing by 2i from the second year
    double targetAmount = 424;     // Target funds in ten thousand

    int yearsNeeded = calculateYearsUntilTarget(initialCapital, annualRate, membershipFee, targetAmount);
    printf("需要的年数: %d\n", yearsNeeded);
    return 0;
}

File: common/StudioFundsCalculator/StudioFundsCalculator.py
def calculate_years_until_target(initial_capital, annual_rate, membership_fee, target_amount):
    if target_amount <= initial_capital:
        return 0  # If the target amount is less than or equal to the initial capital, return 0 years

    current_capital = initial_capital
    year = 0

    while current_capital < target_amount:
        year += 1
        # Calculate end-of-year capital
        current_capital = current_capital * (1 + annual_rate) + membership_fee * year
        print(f"{year}年末 的金额: {current_capital:.2f}")

    return year

if __name__ == "__main__":
    initial_capital = 100  # Initial capital in ten thousand
    annual_rate = 0.10     # Annual interest rate of 10%
    membership_fee = 2     # Membership renewal fee in ten thousand, increasing by 2i from the second year
    target_amount = 424     # Target funds in ten thousand

    years_needed = calculate_years_until_target(initial_capital, annual_rate, membership_fee, target_amount)
    print(f"需要的年数: {years_needed}")